site/en/blog/prerender-pages/index.md

```diff
@@ -0,0 +1,271 @@
+
+From version 3.0.4, the [`web-vitals` library](https://github.com/GoogleChrome/web-vitals) has been updated to handle prerendered navigations in the same way Chrome measure Core Web Vitals. This version also flags prerendered navigations for those metrics in the [`Metric.navigationType`](https://github.com/GoogleChrome/web-vitals#metric) attribute, which is discussed next.
+
+### Measuring prerenders
+
+Whether a page is prerendered can be seen with a non-zero `activationStart` entry of [`PerformanceNavigationTiming`](https://developer.mozilla.org/docs/Web/API/PerformanceNavigationTiming), supplemented with changes for some navigation currently not reported by that. This can then be logged using a Custom Dimension, or similar when logging the page views:
+
+```js
+// Set Custom Dimension for Prerender status
+const prerendered =
+    performance &&
+    performance.getEntriesByType &&
+    performance.getEntriesByType('navigation')[0] &&
+    performance.getEntriesByType('navigation')[0].activationStart > 0;
+gtag('set', {'dimension1': prerendered});
+// Initialise GA - including sending page view by default
+gtag('config', 'UA-12345678-1');
+```
+
+This will allow your analytics to show how many navigation are prerendered compared to other types of navigation, and also allow you to correlation any performance metrics or business metrics to these different navigation types. Faster pages means happier users, which can often have real impact on business measures as our [case studies](https://web.dev/tags/case-study/) show.
+
+As you measure the business impact of prerendering pages for instant navigations, you can decide whether it is worth investing more effort in using this technology to allow more navigations to be prerendered, or to investigate why pages are not being prerendered.
+
+#### Measuring hit rates
+
+As well as measuring the impact of pages that are visited after a prerender, it is also important to measure pages that are prerendered and _not_ subsequently visited. This could imply you are prerendering too much, and using up valuable resources of the user for little benefit.
+
+This is measured by firing an analytics event when speculation rules are inserted. You can then compare the number of these events, to the actual prerender page views. Or alternatively fire another event (`prerender-activated`) on activation if that makes it easier to compare.
+
+The "successful hit rate" can then be calculated by looking at the difference between these two figures. For pages where you are using the Speculation Rules API to prerender the pages, you can adjust the rules appropriately to ensure you keep a high hit rate to maintain the balance between using up the users resources to help them, versus using it needlessly.
```

Comment:
I'm still not sure this can be calculated in enough cases to make this accurate. E.g. if the analytics script doesn't finish downloading before the prerender is discarded, then this won't work.

Also, if we're recommending that analytics vendors defer sending any code until after activation, then this will not work if they end up making that change.

Reply:
I'm suggesting sending an event on the _calling page_, when the prerender speculation rule is added.
Then having another event on the prerendered page after it's activated.
Then measuring the difference.

That's what I've implemented in: https://github.com/GoogleChrome/web.dev/pull/9100 (though using the standard page view for the "after activation" one, since we measure navigation type already).

The only thing is this will also count prerenders from the omnibox, but that's covered by the next paragraph:

> Be aware that some prerendering may be taking place due to the address bar prerendering and not just your speculation rules. You can check the `document.referrer` (which will be blank for address bar navigation including prerendered address bar navigations) if you wish to differentiate these.

Reply:
> I'm suggesting sending an event on the _calling page_, when the prerender speculation rule is added.

Gotcha, that wasn't immediately clear to me, (possibly because I'd read the other draft), but even so I think you need to either remove this section or be much more clear about the limitations of this approach. For example, there's no guarantee that adding speculation rules will trigger prerendering.

Reply:
I added some further clarifications in https://github.com/GoogleChrome/developer.chrome.com/pull/4365/commits/4efcaa23f01cb4d77038a4535c07345f2b69b526.

I'd quite like to keep this, as think we should be advising developers to measure this to avoid needless prerenders.